dev/import-beats/common.go

```diff
@@ -0,0 +1,136 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// Source code origin:
+// github.com/elastic/beats/libbeat/common/mapstr.go
+
+var (
+	// errKeyNotFound indicates that the specified key was not found.
+	errKeyNotFound = errors.New("key not found")
+)
+
+type mapStr map[string]interface{}
+
+// getValue gets a value from the map. If the key does not exist then an error
+// is returned.
+func (m mapStr) getValue(key string) (interface{}, error) {
+	_, _, v, found, err := mapFind(key, m, false)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, errKeyNotFound
+	}
+	return v, nil
+}
+
+// put associates the specified value with the specified key. If the map
+// previously contained a mapping for the key, the old value is replaced and
+// returned. The key can be expressed in dot-notation (e.g. x.y) to put a value
+// into a nested map.
+//
+// If you need insert keys containing dots then you must use bracket notation
+// to insert values (e.g. m[key] = value).
+func (m mapStr) put(key string, value interface{}) (interface{}, error) {
```

Comment:
@urso We should really put mapStr in its own packages. I also had other uses in the past for it.